packages/dds/tree/src/test/simple-tree/utils.ts

```diff
@@ -43,3 +44,122 @@ export function pretty(arg: unknown): number | string {
 	}
 	return JSON.stringify(arg);
 }
+
+const schemaFactory = new SchemaFactory("test");
+
+/**
+ * @internal
+ */
+export const deepTreeNode = schemaFactory.objectRecursive("deep-tree-node", {
+	node: schemaFactory.optionalRecursive([() => DeepTreeNode]),
+	value: schemaFactory.optional(schemaFactory.number),
+});
+
+/**
+ * @internal
+ */
+export class DeepTreeNode extends deepTreeNode {}
+
+/**
+ * Make a deep tree with a single leaf value
+ */
+export function makeDeepTree(depth: number, leafValue: number): DeepTreeNode {
+	return depth === 1
+		? new DeepTreeNode({ node: undefined, value: leafValue })
+		: new DeepTreeNode({ node: makeDeepTree(depth - 1, leafValue), value: undefined });
+}
+
+/**
+ * generate a deep tree with a single leaf value
+ */
+export function generateDeepSimpleTree(depth: number, leafValue: number): DeepTreeNode {
+	if (depth === 0) {
+		return new DeepTreeNode({ node: undefined, value: undefined });
+	}
+
+	return hydrate(DeepTreeNode, makeDeepTree(depth, leafValue));
+}
+
+/**
+ * read the deep tree and return the depth and value of the leaf node
+ */
+export function readDeepSimpleTree(tree: DeepTreeNode): {
+	depth: number;
+	value: number;
+} {
+	let currentNode = tree;
+	let depth = 1;
+
+	while (currentNode.node !== undefined) {
+		depth += 1;
+		currentNode = currentNode.node;
+	}
+
+	const value = currentNode.value ?? 0;
+	return { depth, value };
+}
+
+/**
+ * Upadte the deep tree with a new value
+ */
+export function writeDeepSimpleTree(tree: DeepTreeNode, newValue: number): void {
+	let currentNode = tree;
+
+	while (currentNode.node !== undefined) {
+		currentNode = currentNode.node;
+	}
+
+	currentNode.value = newValue;
+}
+
+/**
+ * @internal
+ */
+export const wideTreeNode = schemaFactory.objectRecursive("wide-tree-node", {
+	values: schemaFactory.array(schemaFactory.number),
+});
+
+/**
+ * @internal
+ */
+export class WideTreeNode extends wideTreeNode {}
+
+/**
+ * Make a wide tree with a single layer
+ */
+export function makeWideTree(width: number, leafValue: number): WideTreeNode {
+	const values = [];
+	for (let i = 0; i < width; i++) {
+		values.push(leafValue);
+	}
+	return new WideTreeNode({ values });
+}
+
+/**
+ * generate a wide tree with a single layer
+ */
+export function generateWideSimpleTree(width: number, leafValue: number): WideTreeNode {
+	return hydrate(WideTreeNode, makeWideTree(width, leafValue));
+}
+
+/**
+ * read the wide tree and return the width and value of the leaf node
+ */
+export function readWideSimpleTree(tree: WideTreeNode): {
+	nodesCount: number;
+	sum: number;
+} {
+	const nodesCount = tree.values.length;
+	let sum = 0;
+	for (const value of tree.values) {
+		sum += value;
+	}
+	return { nodesCount, sum };
+}
+
+/**
+ * write the wide tree with a new value
+ */
+export function writeWideSimpleTree(tree: WideTreeNode, newValue: number): WideTreeNode {
+	return generateWideSimpleTree(tree.values.length, newValue);
+}
```

Comment:
This is not writing to the tree, it's creating a new one. I think this function should do something like `tree.values[tree.values.length-1] = newValue`? And maybe rename it to `writeLastNodeOfWideSimpleTree` to make it super clear that's what it would do.

Reply:
I believe array nodes implement the `IReadonlyArray` interface, but they do allow editing. They allow inserting, removing, and setting (at specific indexes). The functionality is declared here:
https://github.com/microsoft/FluidFramework/blob/main/packages/dds/tree/src/simple-tree/arrayNode.ts#L57

Reply:
I see! Then remove() and insert() to update value here